File: scripts/templates/typescript/part1.ts
import * as fs from 'fs';
import * as path from 'path';

function readInput(filePath: string): string[] {
  return fs.readFileSync(filePath, 'utf8').split('\n').filter(line => line.length > 0);
}

function solve(input: string[]): string {
  // Your solution for part 1 goes here
  return "solution for part 1";
}

const startTime = new Date().getTime();
const inputFile = process.argv.length > 2 ? process.argv[2] : 'input.txt';
const input = readInput(inputFile);
const solution = solve(input);
const endTime = new Date().getTime();

console.log(`Solution: ${solution}`);
console.log(`Completed in ${(endTime - startTime) / 1000} seconds`);

File: scripts/templates/ruby/part1.rb
require 'time'

def read_input(file_path)
  File.readlines(file_path, chomp: true).filter { |line| !line.empty? }
end

def solve(input)
  # Your solution for part 1 goes here
  "solution for part 1"
end

start_time = Time.now
input_file = ARGV.length > 0 ? ARGV[0] : 'input.txt'
input = read_input(input_file)
solution = solve(input)
end_time = Time.now

puts "Solution: #{solution}"
puts "Completed in #{end_time - start_time} seconds"
